### PR TITLE
Typed actor system impl removal

### PR DESCRIPTION
**Pull request by chbatey**
Long day doing all this, i'll review myself tomorrow morning before its worth someone else taking a look.

As discussed in https://github.com/akka/akka/issues/24149

It is worth reviewing the commits separately and leaving them separate in case we want to revert either independently. 

The removal commit removes all (i think) of the machinery for a native typed actor system that we'll hopefully work on once we have akka typed prod ready and it has been adopted or during research time in a branch.

The second commit is a very mechanical switch to wordspec as all the team agreed we shouldn't have two types of tests in the code base.

### CODE REVIEW

**Comment by chbatey** [comment]
PLS BUILD

**Comment by chbatey** [comment]
This is ready for review when you have the time / energy @patriknw :)

File: akka-actor-typed-tests/src/test/scala/docs/akka/typed/IntroSpec.scala [hunk 1]
```diff
@@ -118,22 +118,24 @@ class IntroSpec extends TypedSpec {
...
         val chatRoom = ctx.spawn(ChatRoom.behavior, "chatroom")
         val gabblerRef = ctx.spawn(gabbler, "gabbler")
         ctx.watch(gabblerRef)
-        chatRoom ! GetSession("ol’ Gabbler", gabblerRef)
 
-        Actor.immutable[akka.NotUsed] {
-          (_, _) ⇒ Actor.unhandled
+        Actor.immutablePartial[String] {
+          case (_, "go") ⇒
+            chatRoom ! GetSession("ol’ Gabbler", gabblerRef)
```
Comment (patriknw) [inline akka-actor-typed-tests/src/test/scala/docs/akka/typed/IntroSpec.scala]: why was this change needed?

File: akka-actor-typed/src/main/scala/akka/actor/typed/ActorRef.scala [hunk 2]
```diff
@@ -66,22 +64,10 @@ object ActorRef {
    * Create an ActorRef from a Future, buffering up to the given number of
    * messages in while the Future is not fulfilled.
    */
-  def apply[T](f: Future[ActorRef[T]], bufferSize: Int = 1000): ActorRef[T] =
+  private[akka] def apply[T](f: Future[ActorRef[T]], bufferSize: Int = 1000): ActorRef[T] =
```
Comment (patriknw) [inline akka-actor-typed/src/main/scala/akka/actor/typed/ActorRef.scala]: add the full internal markers
```
/**
   * INTERNAL API
   */
  @InternalApi private[akka] 
```

File: build.sbt [hunk 3]
```diff
@@ -445,3 +445,11 @@ def akkaModule(name: String): Project =
     .settings(akka.AkkaBuild.defaultSettings)
     .settings(akka.Formatting.formatSettings)
     .enablePlugins(BootstrapGenjavadoc)
+
+lazy val typedTests = taskKey[Unit]("Runs all the typed tests")
```
Comment (patriknw) [inline build.sbt]: nice

**Review by patriknw — COMMENTED** [review]
looked briefly at the test rewrite also, good

**Review by patriknw — APPROVED** [review]
LGTM

File: akka-actor-typed-tests/src/test/scala/docs/akka/typed/IntroSpec.scala (hunk 1 above)
Comment (chbatey) [reply akka-actor-typed-tests/src/test/scala/docs/akka/typed/IntroSpec.scala]: https://github.com/akka/akka/issues/24172

File: akka-actor-typed/src/main/scala/akka/actor/typed/ActorRef.scala (hunk 2 above)
Comment (chbatey) [reply akka-actor-typed/src/main/scala/akka/actor/typed/ActorRef.scala]: Updated

File: akka-actor-typed-tests/src/test/scala/akka/actor/typed/ActorContextSpec.scala [hunk 4]
```diff
@@ -373,302 +372,337 @@ class ActorContextSpec extends TypedSpec(ConfigFactory.parseString(
...
+      sync(setup("ctx05", Some(Actor.supervise(_).onFailure(SupervisorStrategy.restart))) { (ctx, startWith) ⇒
         val self = ctx.self
-        val ex = new Exception("KABOOM06")
+        val ex = new Exception("KABOOM05")
         startWith
           .stimulate(_ ! BecomeInert(self), _ ⇒ BecameInert)
-          .stimulate(_ ! Ping(self), _ ⇒ Pong2).keep { subj ⇒
-            muteExpectedException[Exception]("KABOOM06", occurrences = 1)
+          .stimulate(_ ! Ping(self), _ ⇒ Pong2) { subj ⇒
+            val log = muteExpectedException[Exception]("KABOOM05")
```
Comment (johanandren) [inline akka-actor-typed-tests/src/test/scala/akka/actor/typed/ActorContextSpec.scala]: Can't see that log is used anywhere?

File: akka-actor-typed-tests/src/test/scala/akka/actor/typed/RestarterSpec.scala [hunk 5]
```diff
@@ -214,7 +216,7 @@ class RestarterSpec extends TypedSpec {
       inbox.receiveMsg() should ===(GotSignal(PostStop))
     }
 
-    def `must create underlying deferred behavior immediately`(): Unit = {
+    "must create underlying deferred behavior immediately" in {
```
Comment (ktoso) [inline akka-actor-typed-tests/src/test/scala/akka/actor/typed/RestarterSpec.scala]: glad to move to the usual style :)

File: akka-actor-typed/src/main/scala/akka/actor/typed/ActorRef.scala (hunk 2 above)
Comment (ktoso) [reply akka-actor-typed/src/main/scala/akka/actor/typed/ActorRef.scala]: Does not seem to have been updated? Missing InternalApi annotation and comment

File: akka-testkit-typed/src/test/scala/akka/typed/testkit/EffectfulActorContextSpec.scala [hunk 6]
```diff
@@ -80,7 +80,7 @@ object EffectfulActorContextSpec {
 
 class EffectfulActorContextSpec extends FlatSpec with Matchers {
 
-  private val props = Props.empty.withMailboxCapacity(10)
```
Comment (ktoso) [inline akka-testkit-typed/src/test/scala/akka/typed/testkit/EffectfulActorContextSpec.scala]: was this not working on the adapted one? Does this mean we lose the concept of dropping or how it was called there (the mailbox full, it would not be dead letter but that other thing)?

File: akka-actor-typed/src/main/scala/akka/actor/typed/internal/adapter/ActorSystemAdapter.scala [hunk 7]
```diff
@@ -25,15 +25,19 @@ import akka.typed.EventStream
...
   extends ActorSystem[T] with ActorRef[T] with internal.ActorRefImpl[T] with ExtensionsImpl {
 
+  loadExtensions()
+
   import ActorRefAdapter.sendSystemMessage
 
   // Members declared in akka.actor.typed.ActorRef
   override def tell(msg: T): Unit = {
-    if (msg == null) throw new InvalidMessageException("[null] is not an allowed message")
+    if (msg == null) throw InvalidMessageException("[null] is not an allowed message")
```
Comment (ktoso) [inline akka-actor-typed/src/main/scala/akka/actor/typed/internal/adapter/ActorSystemAdapter.scala]: new should be quicker though, ignore warning?

File: akka-actor-typed/src/main/scala/akka/actor/typed/internal/ExtensionsImpl.scala [hunk 8]
```diff
@@ -25,7 +25,7 @@ trait ExtensionsImpl extends Extensions { self: ActorSystem[_] ⇒
   /**
    * Hook for ActorSystem to load extensions on startup
    */
-  protected final def loadExtensions(): Unit = {
+  private[akka] def loadExtensions(): Unit = {
```
Comment (ktoso) [inline akka-actor-typed/src/main/scala/akka/actor/typed/internal/ExtensionsImpl.scala]: annotations please

Comment (ktoso) [reply akka-actor-typed/src/main/scala/akka/actor/typed/internal/ExtensionsImpl.scala]: And comment -- Java people won't see that it's private at all, it looks like a public method to them

File: akka-actor-typed/src/main/scala/akka/actor/typed/ActorSystem.scala [hunk 9]
```diff
@@ -205,26 +199,22 @@ object ActorSystem {
...
-                 classLoader:         Option[ClassLoader]      = None,
-                 executionContext:    Option[ExecutionContext] = None,
-                 actorSystemSettings: ActorSystemSetup         = ActorSystemSetup.empty): ActorSystem[T] = {
-
-    // TODO I'm not sure how useful this mode is for end-users. It has the limitation that untyped top level
-    // actors can't be created, because we have a custom user guardian. I would imagine that if you have
-    // a system of both untyped and typed actors (e.g. adding some typed actors to an existing application)
-    // you would start an untyped.ActorSystem and spawn typed actors from that system or from untyped actors.
-    // Same thing with `wrap` below.
+  private def createInternal[T](name: String, guardianBehavior: Behavior[T],
```
Comment (ktoso) [inline akka-actor-typed/src/main/scala/akka/actor/typed/ActorSystem.scala]: good :)

File: akka-actor-typed-tests/src/test/scala/akka/actor/typed/PropsSpec.scala [hunk 10]
```diff
@@ -5,35 +5,16 @@ package akka.actor.typed
...
 
-  val dispatcherFirst = DispatcherDefault(MailboxCapacity(666, DispatcherFromConfig("pool")))
-  val mailboxFirst = MailboxCapacity(999) withNext dispatcherFirst
+  val dispatcherFirst = DispatcherDefault(DispatcherFromConfig("pool"))
 
-  object `A Props` {
+  "A Props" must {
 
-    def `must get first dispatcher`(): Unit = {
+    "must get first dispatcher" in {
```
Comment (johanandren) [inline akka-actor-typed-tests/src/test/scala/akka/actor/typed/PropsSpec.scala]: must must, and the next one as well

File: akka-actor-typed-tests/src/test/scala/akka/actor/typed/RestarterSpec.scala [hunk 11]
```diff
@@ -63,22 +62,25 @@ class RestarterSpec extends TypedSpec {
...
+  import RestarterSpec._
 
-    def mkCtx(behv: Behavior[Command]): EffectfulActorContext[Command] =
-      new EffectfulActorContext("ctx", behv, 1000, system)
+  def mkCtx(behv: Behavior[Command]): EffectfulActorContext[Command] =
+    new EffectfulActorContext("ctx", behv, 1000, system)
 
-    def `must receive message`(): Unit = {
+  "A restarter" must {
+    "must receive message" in {
```
Comment (johanandren) [inline akka-actor-typed-tests/src/test/scala/akka/actor/typed/RestarterSpec.scala]: and more must must

File: akka-actor-typed/src/main/scala/akka/actor/typed/internal/ExtensionsImpl.scala (hunk 8 above)
Comment (chbatey) [reply akka-actor-typed/src/main/scala/akka/actor/typed/internal/ExtensionsImpl.scala]: done

File: akka-actor-typed-tests/src/test/scala/akka/actor/typed/RestarterSpec.scala (hunk 5 above)
Comment (chbatey) [reply akka-actor-typed-tests/src/test/scala/akka/actor/typed/RestarterSpec.scala]: That was a painful switch :)

File: akka-actor-typed/src/main/scala/akka/actor/typed/ActorRef.scala [hunk 12]
```diff
@@ -16,12 +16,10 @@ import scala.util.Success
  * Actor instance. Sending a message to an Actor that has terminated before
  * receiving the message will lead to that message being discarded; such
  * messages are delivered to the [[DeadLetter]] channel of the
- * [[EventStream]] on a best effort basis
+ * [[akka.typed.EventStream]] on a best effort basis
```
Comment (johanandren) [inline akka-actor-typed/src/main/scala/akka/actor/typed/ActorRef.scala]: that wrong package?

File: akka-actor-typed/src/main/scala/akka/actor/typed/ActorSystem.scala [hunk 13]
```diff
@@ -139,11 +139,11 @@ abstract class ActorSystem[-T] extends ActorRef[T] with Extensions { this: inter
    * Ask the system guardian of this system to create an actor from the given
    * behavior and props and with the given name. The name does not need to
    * be unique since the guardian will prefix it with a running number when
-   * creating the child actor. The timeout sets the timeout used for the [[akka.actor.typed.scaladsl.AskPattern$]]
+   * creating the child actor. The timeout sets the timeout used for the [[akka.actor.typed.scaladsl.AskPattern]]
    * invocation when asking the guardian.
    *
    * The returned Future of [[ActorRef]] may be converted into an [[ActorRef]]
-   * to which messages can immediately be sent by using the [[ActorRef$.apply[T](s*]]
+   * to which messages can immediately be sent by using the [[ActorRef.apply[T](s*]]
```
Comment (johanandren) [inline akka-actor-typed/src/main/scala/akka/actor/typed/ActorSystem.scala]: Sure the $'s shouldn't be in these two to disambiguate between objects and traits with the same name?

File: akka-testkit-typed/src/test/scala/akka/typed/testkit/EffectfulActorContextSpec.scala (hunk 6 above)
Comment (chbatey) [reply akka-testkit-typed/src/test/scala/akka/typed/testkit/EffectfulActorContextSpec.scala]: Nothing read it after i removed ActorSystemImpl so i guess not

File: akka-testkit-typed/src/main/scala/akka/typed/testkit/StubbedActorContext.scala [hunk 14]
```diff
@@ -1,14 +1,108 @@
...
+
+import scala.annotation.tailrec
+import scala.util.control.NonFatal
+
+/**
+ * A local synchronous ActorRef that invokes the given function for every message send.
+ * This reference can be watched and will do the right thing when it receives a [[akka.actor.typed.internal.DeathWatchNotification]].
+ * This reference cannot watch other references.
+ */
+private[akka] final class FunctionRef[-T](
```
Comment (johanandren) [inline akka-testkit-typed/src/main/scala/akka/typed/testkit/StubbedActorContext.scala]: Why did this end up in the testkit?

**Review by johanandren — APPROVED** [review]
Added some comments, LGTM with those fixed.

File: akka-actor-typed/src/main/scala/akka/actor/typed/internal/adapter/ActorSystemAdapter.scala (hunk 7 above)
Comment (chbatey) [reply akka-actor-typed/src/main/scala/akka/actor/typed/internal/adapter/ActorSystemAdapter.scala]: You think it'll make a difference? Not on any hot path, is a bug in user code to hit it

File: akka-testkit-typed/src/test/scala/akka/typed/testkit/EffectfulActorContextSpec.scala (hunk 6 above)
Comment (patriknw) [reply akka-testkit-typed/src/test/scala/akka/typed/testkit/EffectfulActorContextSpec.scala]: Bounded mailbox is not a new concept. We could make it default and/or make it simple to specify via Props, but that should be a separate ticket.

File: akka-actor-typed/src/main/scala/akka/actor/typed/internal/adapter/ActorSystemAdapter.scala (hunk 7 above)
Comment (ktoso) [reply akka-actor-typed/src/main/scala/akka/actor/typed/internal/adapter/ActorSystemAdapter.scala]: Yeah hm, I don't mind either I guess ;)

File: akka-testkit-typed/src/test/scala/akka/typed/testkit/EffectfulActorContextSpec.scala (hunk 6 above)
Comment (ktoso) [reply akka-testkit-typed/src/test/scala/akka/typed/testkit/EffectfulActorContextSpec.scala]: Yeap, the new part in Typed was that there was a difference between dropped and deadletters AFAIR which I really liked. Ticket pls :)

File: akka-actor-typed/src/main/scala/akka/actor/typed/ActorRef.scala (hunk 12 above)
Comment (chbatey) [reply akka-actor-typed/src/main/scala/akka/actor/typed/ActorRef.scala]: Well i didn't move it it to akka.actor as the untyped EventStream isn't in there, it could be akka.event.typed.EventStream, WDYT?

File: akka-actor-typed-tests/src/test/scala/akka/actor/typed/ActorContextSpec.scala (hunk 4 above)
Comment (chbatey) [reply akka-actor-typed-tests/src/test/scala/akka/actor/typed/ActorContextSpec.scala]: Pfft making me fix things that were already broken! Okay :)

File: akka-actor-typed-tests/src/test/scala/akka/actor/typed/PropsSpec.scala (hunk 10 above)
Comment (chbatey) [reply akka-actor-typed-tests/src/test/scala/akka/actor/typed/PropsSpec.scala]: Sometimes you have to be certain

File: akka-actor-typed-tests/src/test/scala/akka/actor/typed/RestarterSpec.scala (hunk 11 above)
Comment (chbatey) [reply akka-actor-typed-tests/src/test/scala/akka/actor/typed/RestarterSpec.scala]: fixed all in this test

File: akka-actor-typed/src/main/scala/akka/actor/typed/ActorSystem.scala (hunk 13 above)
Comment (chbatey) [reply akka-actor-typed/src/main/scala/akka/actor/typed/ActorSystem.scala]: :+1:

File: akka-testkit-typed/src/main/scala/akka/typed/testkit/StubbedActorContext.scala (hunk 14 above)
Comment (chbatey) [reply akka-testkit-typed/src/main/scala/akka/typed/testkit/StubbedActorContext.scala]: It used to be used in the old impl, now it is only used in the stubbed actor context. I guess we can re-evaluate how that works during the testkit issues for typed.

File: akka-actor-typed/src/main/scala/akka/actor/typed/ActorRef.scala (hunk 12 above)
Comment (johanandren) [reply akka-actor-typed/src/main/scala/akka/actor/typed/ActorRef.scala]: sounds good given the other package names

Comment (chbatey) [reply akka-actor-typed/src/main/scala/akka/actor/typed/ActorRef.scala]: Renamed

File: akka-testkit-typed/src/main/scala/akka/typed/testkit/StubbedActorContext.scala (hunk 14 above)
Comment (johanandren) [reply akka-testkit-typed/src/main/scala/akka/typed/testkit/StubbedActorContext.scala]: okok

**Comment by chbatey** [comment]
Was a known failure

**Comment by chbatey** [comment]
PLS BUILD

**Comment by chbatey** [comment]
PLS BUILD
